### PR TITLE
Packager config mods

### PR DESCRIPTION
**Pull request by banaag**
Changes to amppackager toml file to accommodate ACME config changes.

Addresses #93

### CODE REVIEW

**Comment by twifkak** [comment]
@googlebot

File: packager/util/config.go [hunk 1]
```diff
@@ -24,13 +24,17 @@ import (
...
-	LocalOnly bool
-	Port      int
-	CertFile  string // This must be the full certificate chain.
-	KeyFile   string // Just for the first cert, obviously.
-	OCSPCache string
+	LocalOnly	bool
+	Port		int
+	CertFile	string // This must be the full certificate chain.
+	KeyFile		string // Just for the first cert, obviously.
+	NewCertFile	string // The new full certificate chain replacing the expired one.
```
Comment (twifkak) [inline packager/util/config.go]: Can you explain the semantics here? Currently, CertFile is only read by packager. Is NewCertFile read/write? What happens in 180-days with NewNewCertFile? Or is CertFile now read/write, with some process moving NewCertFile to CertFile before it's time for NewNew?

File: packager/util/config.go [hunk 2]
```diff
@@ -24,13 +24,17 @@ import (
...
+	NewKeyFile	string // For the first cert in NewCertFile.
```
Comment (twifkak) [inline packager/util/config.go]: I don't think NewKeyFile is necessary for v1. (Correct me if I'm wrong!) It looks like ACME does have a key change protocol (https://tools.ietf.org/html/rfc8555#section-7.3.5) but the usual 90-day renewal won't involve it AFAIK.

File: packager/util/config.go [hunk 3]
```diff
@@ -24,13 +24,17 @@ import (
...
+	AutoRenewCert	bool   // Should we auto-renew cert? Defaults to false.
```
Comment (twifkak) [inline packager/util/config.go]: (opt) Could we simplify? Auto-renew if and only if there's a non-empty AccountURL?

File: packager/util/config.go [hunk 4]
```diff
@@ -50,6 +54,16 @@ type URLPattern struct {
 	SamePath               *bool
 }
 
+type ACMEConfig struct {
+	Prod	*ACMEServerConfig
+	Staging	*ACMEServerConfig
```
Comment (twifkak) [inline packager/util/config.go]: I'd rename to Development; matches our `-development` command-line flag. We have plans for a `-staging` flag (#188) but I think that'll use production config in every way except the cert-url it writes into the SXG.

File: packager/util/config.go (hunk 1 above)
Comment (banaag) [reply packager/util/config.go]: NewCertFile will be read/write.  I envision that both CertFile and NewCertFile will both be set when both are valid (overlap) and that once CertFile becomes invalid, NewCertFile will replace it (CertFile = NewCertFile) and NewCertFile will be empty.  I would think that this applies to the disk copies as well (which may need to be some sort of shared disk, if multiple replicas of amppackager are running).

File: packager/util/config.go (hunk 3 above)
Comment (banaag) [reply packager/util/config.go]: That sounds reasonable.  I will adjust the comments to reflect that decision.

File: packager/util/config.go (hunk 4 above)
Comment (banaag) [reply packager/util/config.go]: Development sounds reasonable.

File: packager/util/config.go (hunk 2 above)
Comment (banaag) [reply packager/util/config.go]: You are correct, I will remove the field.

File: packager/util/config.go (hunk 1 above)
Comment (twifkak) [reply packager/util/config.go]: OK, SGTM! Thanks for clarifying.

File: packager/util/config.go [hunk 5]
```diff
@@ -28,9 +28,13 @@ type Config struct {
 	Port		int
 	CertFile	string // This must be the full certificate chain.
 	KeyFile		string // Just for the first cert, obviously.
+
+	// NewCertFile will be read/write. CertFile and NewCertFile will be set when both
```
Comment (twifkak) [inline packager/util/config.go]: How about for the first sentence, "When set, both CertFile and NewCertFile will be read/write." (Though, as you're probably aware, the user-facing documentation is in amppkg.example.toml.)

Comment (banaag) [reply packager/util/config.go]: Done.

**Pull request merged**